cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -0,0 +1,656 @@
+										Opa: &authorinov1beta3.OpaAuthorizationSpec{
+											Rego: "allow = false",
+										},
+									},
+									CommonEvaluatorSpec: authorinov1beta3.CommonEvaluatorSpec{
+										Priority: 2,
+									},
+								},
+							},
+						},
+						Response: &kuadrantv1.MergeableResponseSpec{
+							Unauthorized: &kuadrantv1.MergeableDenyWithSpec{
+								DenyWithSpec: authorinov1beta3.DenyWithSpec{
+									Code:    302,
+									Headers: credentialsHeader(pol.Spec.Auth.TokenSource, igw),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+type authPolicyMutateFn func(desired, existing *kuadrantv1.AuthPolicy) bool
+
+func authPolicyMutator(opts ...authPolicyMutateFn) reconcilers.MutateFn {
+	return func(desiredObj, existingObj client.Object) (bool, error) {
+		existing, ok := existingObj.(*kuadrantv1.AuthPolicy)
+		if !ok {
+			return false, fmt.Errorf("%T is not a *kuadrantv1.AuthPolicy", existingObj)
+		}
+		desired, ok := desiredObj.(*kuadrantv1.AuthPolicy)
+		if !ok {
+			return false, fmt.Errorf("%T is not a *kuadrantv1.AuthPolicy", desiredObj)
+		}
+
+		update := false
+
+		// Loop through each option
+		for _, opt := range opts {
+			tmpUpdate := opt(desired, existing)
+			update = update || tmpUpdate
+		}
+
+		return update, nil
+	}
+}
+
+func authPolicyTargetRefMutator(desired, existing *kuadrantv1.AuthPolicy) bool {
+	update := false
+
+	if !reflect.DeepEqual(existing.Spec.TargetRef, desired.Spec.TargetRef) {
+		existing.Spec.TargetRef = desired.Spec.TargetRef
+		update = true
+	}
+
+	return update
+}
+
+func mainAuthPolicyAuthenticationMutator(desired, existing *kuadrantv1.AuthPolicy) bool {
+	update := false
+
+	if !reflect.DeepEqual(existing.Spec.Overrides.AuthScheme.Authentication["oidc"], desired.Spec.Overrides.AuthScheme.Authentication["oidc"]) {
+		existing.Spec.Overrides.AuthScheme.Authentication["oidc"] = desired.Spec.Overrides.AuthScheme.Authentication["oidc"]
+		update = true
+	}
+
+	return update
+}
+
+func mainAuthPolicyAuthorizationMutator(desired, existing *kuadrantv1.AuthPolicy) bool {
+	update := false
+
+	if !reflect.DeepEqual(existing.Spec.Overrides.AuthScheme.Authorization, desired.Spec.Overrides.AuthScheme.Authorization) {
+		existing.Spec.Overrides.AuthScheme.Authorization = desired.Spec.Overrides.AuthScheme.Authorization
+		update = true
+	}
+
+	return update
+}
+
+func credentialsSource(tokenSource authorinov1beta3.Credentials) authorinov1beta3.Credentials {
+	switch tokenSource.GetType() {
+	case authorinov1beta3.AuthorizationHeaderCredentials:
+		return authorinov1beta3.Credentials{AuthorizationHeader: &authorinov1beta3.Prefixed{Prefix: tokenSource.AuthorizationHeader.Prefix}}
+	case authorinov1beta3.CustomHeaderCredentials:
+		return authorinov1beta3.Credentials{CustomHeader: &authorinov1beta3.CustomHeader{Named: authorinov1beta3.Named{Name: tokenSource.CustomHeader.Name}}}
+	case authorinov1beta3.QueryStringCredentials:
```

Comment:
These cases are not working at the moment, since we are relying on the metadata phase of authorino's pipeline to store the token in cookies at callback time... we'll need to find a solution either finding a way to keep this token around or normalizing it with a wristband token maybe (?)